File: minStack/minStack.py
class MinStack:

    def __init__(self):
        """
        initialize your data structure here.
        """
        self.s = []

    def push(self, x):
        """
        :type x: int
        :rtype: None
        """
        self.s.append(x)
        

    def pop(self):
        """
        :rtype: None
        """
        self.s.pop()

    def top(self):
        """
        :rtype: int
        """
        if len(self.s) == 0:return None
        return self.s[len(self.s)-1]

    def getMin(self):
        """
        :rtype: int
        """
        if len(self.s)==0:return 0
        minn = self.s[0]
        for n in self.s:
            minn = min(minn,n)
        return minn

        


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(x)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()

File: minStack/minStack.java
class MinStack {
    private List<Integer> s;

    /** initialize your data structure here. */
    public MinStack() {
        s = new ArrayList<Integer>();
    }
    
    public void push(int x) {
        s.add(x);
    }
    
    public void pop() {
        s.remove(s.size()-1);
    }
    
    public int top() {
        return s.get(s.size()-1);
    }
    
    public int getMin() {
        int minn = s.get(0);
        for(int i=1;i<s.size();i++){
            minn = Math.min(minn,s.get(i));
        }
        return minn;
    }
}

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack obj = new MinStack();
 * obj.push(x);
 * obj.pop();
 * int param_3 = obj.top();
 * int param_4 = obj.getMin();
 */